sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -342,13 +366,28 @@ public ResourceId apply(String input) {
     }
 
     /**
+     * Set the base directory used to generate temporary files.
+     */
+    @Experimental(Kind.FILESYSTEM)
+    public Write<T> withTempDirectory(ValueProvider<ResourceId> tempDirectory) {
+      return toBuilder().setTempDirectory(tempDirectory).build();
+    }
+
+    /**
      * Configures the {@link FileBasedSink.FilenamePolicy} that will be used to name written files.
      */
     public Write<T> withFilenamePolicy(FilenamePolicy filenamePolicy) {
       return toBuilder().setFilenamePolicy(filenamePolicy).build();
     }
 
     /**
+     * Use a {@link DynamicDestinations} object to vend {@link FilenamePolicy} objects. These
+     * objects can
```

Comment:
Removed DynamicDestinations from AvroIO as it was incomplete and will go in a separate PR (to include dynamic schemas)